build-tools/packages/build-cli/src/commands/generate/typetests.ts

```diff
@@ -155,13 +152,16 @@ export default class GenerateTypetestsCommand extends PackageCommand<
  * Generated by flub generate:typetests in @fluid-tools/build-cli.
  */
 
+import type { TypeOnly, MinimalType, FullType } from "@fluidframework/build-tools";
+
 import type * as old from "${previousPackageName}${
 				previousPackageLevel === ApiLevel.public ? "" : `/${previousPackageLevel}`
 			}";
 
 import type * as current from "../../index.js";
-		`.trim(),
-			typeOnly,
+
+export type MakeUnusedImportErrorsGoAway = TypeOnly<0> |  MinimalType<0> | FullType<0> | typeof old | typeof current;
```

Comment:
I feel the same. I made tree's tsconfig work the other way so its only lint errors (and only when name is not prefixed with "_"). Might be worth applying that approach more generally, but for now this hack is more comparable.

Reply:
We should probably file something to track updating the repo to use linting to check for this instead of tsc, and an item to come back and clean this up once that's done.

I 100% agree with making this check a linter responsibility and not a tsc one.

Reply:
making this a declare instead of an export made this a little cleaner. Still silly to have to do.